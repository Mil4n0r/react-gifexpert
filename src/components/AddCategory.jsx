import { PropTypes } from 'prop-types';
import React, { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {
  
  const [inputValue, setInputValue] = useState('');
  
  const onInputChange = ({target:{value}}) => {
    setInputValue(value)
  }

  const onSubmit = (event) => {
    event.preventDefault();
    console.log(inputValue)
    if(inputValue.trim().length <= 1) {
      console.log("OOO")
      return;
    }
    setInputValue('');
    onNewCategory( inputValue.trim() );
  }

  return (
    <form onSubmit={ (event) => onSubmit(event) } aria-label="form">
      <input
        type="text" 
        placeholder='Buscar gifs'
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}