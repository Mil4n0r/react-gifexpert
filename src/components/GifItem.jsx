//xport const GifItem = ({image:{title, url}}) => {
export const GifItem = ({title, url, id}) => {
  
  return (
    <div className="card">
      <img src={ url } alt={ title } />
      <p>{ title }</p>
    </div>
  )
}
