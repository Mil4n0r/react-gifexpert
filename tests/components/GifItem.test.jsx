import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Comprobaci√≥n del componente <GifItem />", () => {
  const title = "Saitama";
  const url = "https://ejemplo.com/ejemplo.jpg";

  test("should match the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show an image with the right URL and ALT", () => {
    render(<GifItem title={title} url={url} />);
    //screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').src).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should show the title in the component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
