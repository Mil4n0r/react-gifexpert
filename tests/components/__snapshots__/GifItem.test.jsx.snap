// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comprobaci√≥n del componente <GifItem /> should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://ejemplo.com/ejemplo.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
